pkg/server/commands/write_assertions.go

```diff
@@ -66,8 +66,13 @@ func (w *WriteAssertionsCommand) Execute(ctx context.Context, req *openfgav1.Wri
 		}
 
 		for _, ct := range assertion.GetContextualTuples().GetTupleKeys() {
-			if err = validation.ValidateUserObjectRelation(typesys,
-				tupleUtils.NewTupleKey(ct.GetObject(), ct.GetRelation(), ct.GetUser())); err != nil {
+			if err = validation.ValidateUserObjectRelationCondition(typesys,
+				tupleUtils.NewTupleKeyWithCondition(
+					ct.GetObject(),
+					ct.GetRelation(),
+					ct.GetUser(),
+					ct.GetCondition().GetName(),
+					ct.GetCondition().GetContext())); err != nil {
```

Comment:
is there the possibility of `ct.GetCondition()` returning `nil` here and causing a panic?

Reply:
No, gRPC checks and return nil for both "GetCondition" and "GetName or GetContext"